presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/StandardReadMappings.java

```diff
@@ -126,9 +128,9 @@ public static ReadMapping dateReadMapping()
         return longReadMapping(DATE, (resultSet, columnIndex) -> {
             // JDBC returns a date using a timestamp at midnight in the JVM timezone
             long localMillis = resultSet.getDate(columnIndex).getTime();
-            // Convert it to a midnight in UTC
+            // Convert it to a ~midnight in UTC. This won't be precisely midnight if JVM zone had a 'forward' DST change on that particular date, exactly at midnight.
             long utcMillis = ISOChronology.getInstance().getZone().getMillisKeepLocal(UTC, localMillis);
-            // convert to days
+            // convert to days (this also truncates excess millis if utcMillis isn't exactly midnight)
             return MILLISECONDS.toDays(utcMillis);
```

Comment:
Only this computation could be replaced with very simple and neat:
```
return resultSet.getObject(columnIndex, LocalDate.class).toEpochDays();
```

except that it doesn't work for redshift (old postgres driver) and sql server (even if we upgrade to newest driver).